src/Codec.elm

```diff
@@ -235,6 +236,10 @@ composite enc dec (Codec codec) =
 
 
 {-| Represents an optional value.
+
+If the value isn't present, is `null`, or has the wrong type, the decoder will return `Nothing`.
+If the value is `Nothing` then the resulting object will contain the field with a `null` value.
```

Comment:
This can be used in lists and at the top level, so talking about fields only is imprecise
Also, I'm not sure if it's better to talk about `type` or `shape`... ideas?